src/templates/healthcert/parsers/vacHealthCertV1/generateMultiQrSection.tsx

```diff
@@ -1,65 +1,69 @@
+interface GroupedOfflineQr {
+  manufacturer: string;
+  signedEuHealthCerts: notarise.SignedEuHealthCert[];
+}
 
-export const generateMultiQrSection = (document: NotarisedHealthCert): JSX.Element => {
-  const patient = document.fhirBundle.entry.find(entry => entry.resourceType === "Patient") as Patient;
-  const patientName = typeof patient?.name?.[0] === "object" ? patient?.name[0]?.text : "";
-  const url = (document.notarisationMetadata as any)?.url;
-  const signedEuHealthCert = (document.notarisationMetadata as any)?.signedEuHealthCert;
+const OfflineQrGroup: React.FC<{
+  groupedOfflineQr: GroupedOfflineQr;
+}> = ({ groupedOfflineQr }) => {
+  return (
+    <>
+      <QrRowCenter>
+        <Bold>{groupedOfflineQr.manufacturer}</Bold>
+        {groupedOfflineQr.signedEuHealthCerts.map((signedEuHealthCert, i) => (
+          <QrColCenter key={i}>
+            <QrCodeContainerWithBorder key={i}>
+              <QRCode value={signedEuHealthCert.qr} level={"M"} size={200} />
+            </QrCodeContainerWithBorder>
+            DOSE {signedEuHealthCert.dose}
+          </QrColCenter>
+        ))}
+      </QrRowCenter>
+      <br />
+    </>
+  );
+};
+
+const generateOfflineQrSection = (document: NotarisedHealthCert): JSX.Element[] => {
+  const signedEuHealthCerts = document.notarisationMetadata.signedEuHealthCerts || [];
+  const groupedEuHealthCerts = groupBy(signedEuHealthCerts, "vaccineCode");
+  const groupedEuHealthCertKeys = Object.keys(groupedEuHealthCerts);
 
+  return groupedEuHealthCertKeys.map((vaccineCode, i) => {
+    const immunization = document.fhirBundle.entry.find(
+      entry => entry.resourceType === "Immunization" && entry.vaccineCode.coding[0]?.code === vaccineCode
+    ) as Immunization;
+    const groupedOfflineQr = {
+      manufacturer: immunization?.vaccineCode?.coding[0]?.display || "",
+      signedEuHealthCerts: groupedEuHealthCerts[vaccineCode]
+    };
+    return <OfflineQrGroup key={i} groupedOfflineQr={groupedOfflineQr} />;
+  });
+};
+
+export const generateMultiQrSection = (document: NotarisedHealthCert): JSX.Element => {
+  const offlineQrSection = generateOfflineQrSection(document);
   return (
     <>
       <ResultSection>
-        <p>Based on Singapore&apos;s requirement, {patientName} has been fully vaccinated against COVID-19.</p>
-      </ResultSection>
-      <TravellerInfoSection>
-        Note: Travellers are subject to the country or region&apos;s requirements prior to travel.
-        <br />
-        The QR code used for verification is based on your <u>destination country</u>.
-      </TravellerInfoSection>
-      <br />
-      <QrRow>
-        <QrCol info>
-          <Bold>Offline QR Verification</Bold>
-          <br />
+        <Bold>Offline QR Verification</Bold>
+        <p>
           This QR Code does not require an internet connection to verify. Currently only the European Union (EU)
           supports this option of verification.
           <br />
```

Comment:
If possible, use <p> or css style, and avoid <br/>
https://stackoverflow.com/a/1726103/6514532